### PR TITLE
Adding pre-commit to repo

### PR DESCRIPTION
**Pull request by jessemorris**
- adding pre-commit to requirements.txt
- adding pre-commit config from pycharged up folder
- adding setup and pre-commit to readme.md
- running pre-commit to format python files

### CODE REVIEW

**Comment by auscompgeek** [comment]
I've configured pre-commit.ci on this repo now too

File: requirements.txt [hunk 1]
```diff
@@ -6,4 +6,5 @@ opencv-python
 pytest
 pynetworktables
 robotpy-cscore
-robotpy-wpimath
\ No newline at end of file
+robotpy-wpimath
+pre-commit
```
Comment (auscompgeek) [inline requirements.txt]: We're using this requirements file to dictate what will get installed to the coprocessor. I don't think this should go here.
```suggestion
```

Comment (LucienMorey) [reply requirements.txt]: Jesse tried to clarify this with me, but I didn't even think about it. What do you think is a better place to store it @auscompgeek? Just a secondary requirements file or something different?

Comment (auscompgeek) [reply requirements.txt]: Given there are multiple ways to install `pre-commit`, e.g. `brew` or other system package manager, I'm not sure this should go in a requirements file at all.

Comment (LucienMorey) [reply requirements.txt]: It sounds like removing it entirely is your preferred solution here. is that correct?

Comment (auscompgeek) [reply requirements.txt]: That's probably the pragmatic solution here, yeah.

**Review by LucienMorey — APPROVED** [review]

**Review by auscompgeek — APPROVED** [review]
LGTM when CI is green

**Pull request merged**